go/runtime/host/protocol/connection.go

```diff
@@ -5,23 +5,44 @@ import (
 	"context"
 	"fmt"
 	"net"
+	"reflect"
 	"sync"
+	"time"
 
 	"github.com/opentracing/opentracing-go"
 	opentracingExt "github.com/opentracing/opentracing-go/ext"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/spf13/viper"
 
 	"github.com/oasislabs/oasis-core/go/common"
 	"github.com/oasislabs/oasis-core/go/common/cbor"
 	"github.com/oasislabs/oasis-core/go/common/errors"
 	"github.com/oasislabs/oasis-core/go/common/logging"
 	"github.com/oasislabs/oasis-core/go/common/tracing"
 	"github.com/oasislabs/oasis-core/go/common/version"
+	"github.com/oasislabs/oasis-core/go/oasis-node/cmd/common/metrics"
 )
 
 const moduleName = "rhp/internal"
 
 // ErrNotReady is the error reported when the Runtime Host Protocol is not initialized.
-var ErrNotReady = errors.New(moduleName, 1, "rhp: not ready")
+var (
+	ErrNotReady = errors.New(moduleName, 1, "rhp: not ready")
+
+	rhpLatency = prometheus.NewSummaryVec(
```

Comment:
Could you also add counters for successes and failures as you can get that by just checking if there was an error.